components/server/src/auth/host-context-provider-impl.ts

```diff
@@ -32,33 +34,41 @@ export class HostContextProviderImpl implements HostContextProvider {
 
     async init() {
         await this.ensureInitialized();
+
+        // schedule periodic update of dynamic hosts
+        const updateDynamicHostsJob = repeat(async () => {
+            const span = TraceContext.startSpan("updateDynamicHosts");
+            try {
+                await this.updateDynamicHosts({span});
+            } catch (error) {
+                log.error(`Failed to update dynamic hosts.`, error);
+            } finally {
+                span.finish();
+            }
+        }, 10000);
```

Comment:
@AlexTugarev What's the use-case again where we have to sync this often?

And: can we change the "add host provider" code to:
 - add entry to the DB
 - PLUS execute the "addDynamicHostProvider" from below
so we don't rely on this timeout anymore?

Just saw the calls in honeycomg on staging, and I'd really love to set this to 30s or so, even with the optimization.